spec.html

```diff
@@ -4488,53 +4488,64 @@ <h2>Syntax</h2>
-          <ul>
-            <li>
-              it is not `0`; or
-            </li>
-            <li>
-              there is a non-zero digit to its left and there is a non-zero digit, not in the |ExponentPart|, to its right.
-            </li>
-          </ul>
+        <emu-clause id="sec-runtime-semantics-stringnumericvalue" type="sdo" aoid="StringNumericValue" oldids="sec-runtime-semantics-mv-s">
+          <h1>Runtime Semantics: StringNumericValue</h1>
+          <p>The conversion of a String to a Number value is similar overall to the determination of the Number value for a numeric literal (see <emu-xref href="#sec-literals-numeric-literals"></emu-xref>), but some of the details are different, so the process for converting a String numeric literal to a value of Number type is given here.</p>
+          <emu-grammar>StringNumericLiteral ::: StrWhiteSpace?</emu-grammar>
+          <emu-alg>
+            1. Return *+0*<sub>𝔽</sub>.
+          </emu-alg>
+          <emu-grammar>StringNumericLiteral ::: StrWhiteSpace? StrNumericLiteral StrWhiteSpace?</emu-grammar>
+          <emu-alg>
+            1. Return StringNumericValue of |StrNumericLiteral|.
+          </emu-alg>
+          <emu-grammar>StrNumericLiteral ::: NonDecimalIntegerLiteral</emu-grammar>
+          <emu-alg>
+            1. Return 𝔽(MV of |NonDecimalIntegerLiteral|).
+          </emu-alg>
+          <emu-grammar>StrDecimalLiteral ::: `-` StrUnsignedDecimalLiteral</emu-grammar>
+          <emu-alg>
+            1. Let _a_ be StringNumericValue of |StrUnsignedDecimalLiteral|.
+            1. If _a_ is *+0*<sub>𝔽</sub>, return *-0*<sub>𝔽</sub>.
+            1. Return -_a_.
+          </emu-alg>
+          <emu-grammar>StrUnsignedDecimalLiteral ::: `Infinity`</emu-grammar>
+          <emu-alg>
+            1. Return *+&infin;*<sub>𝔽</sub>.
+          </emu-alg>
+          <emu-grammar>StrUnsignedDecimalLiteral ::: DecimalDigits `.` DecimalDigits? ExponentPart?</emu-grammar>
+          <emu-alg>
+            1. Let _a_ be MV of the first |DecimalDigits|.
+            1. If the second |DecimalDigits| is present, let _b_ be MV of the second |DecimalDigits|. Otherwise, let _b_ be 0.
+            1. If |ExponentPart| is present, let _e_ be MV of |ExponentPart|. Otherwise, let _e_ be 0.
+            1. Let _n_ be the number of code points in the second |DecimalDigits|.
```

Comment:
If `the second |DecimalDigits|` is absent, it's unclear if we can talk about the number of code points in it. (The spec says basically nothing about absent Parse Nodes.) On the other hand, it'd be reasonable to take the number of code points as zero.

I'm just noting this in case the editors want to mull it over.

Reply:
I would prefer to be more explicit, as in

```suggestion
            1. If the second |DecimalDigits| is present, let _n_ be the number of code points in the second |DecimalDigits|. Otherwise, let _n_ be 0.
```

Also I'd move it up a line, at which point it could be combined with what would then be the previous line, as in

```
1. If the second |DecimalDigits| is present, then
  1. Let _b_ be MV of the second |DecimalDigits|.
  1. Let _n_ be the number of code points in the second |DecimalDigits|.
1. Else,
  1. Let _b_ be 0.
  1. Let _n_ be 0.
```